### DISCUSSION TITLE
NUCLEO_F429ZI. Is it possible to use CAN2 and Etnernet simultaneously?

### DISCUSSION THREAD

**Discussion by alsor62** [micropython/micropython#12544] [STM32 / Pyboard] [answered]
hi,
for devboard NUCLEO_F429ZI are specified in mpconfigboard.h at the same time
#define MICROPY_HW_CAN2_TX (pin_B13)
And
#define MICROPY_HW_ETH_RMII_TXD1 (pin_B13)
Does this mean I can't use CAN2 and Ethernet at the same time?

**Comment by nspsck** [comment, marked as answer]
Hi,

As for this implementation, no, you can not use them at the same time.

But there is a good news. According to STM32CubeMX, CAN2 and ETH (RMII mode) can be used at the same time.
![image](https://github.com/micropython/micropython/assets/86347287/7bd51797-ba96-46ef-a56a-62ce4fc46f19)
You just have to set the pins accordingly, i.e. use `pin_G14` for `ETH_RMII_TXD1`.
Or you can change the `CAN2_TX` to `pin_B6`.

**Reply by alsor62** [reply]
Thanks for the answer.
But to do this, you need to specify the corresponding pins in the mpconfigboard.h file and build micropython?

**Reply by nspsck** [reply]
Hi, I don't think that you need to specify the pins. The pin name should be already defined in pin.csv.

**Reply by alsor62** [reply]
OK. Thanks for the answer. But it is necessary to rebuild micropython.

**Reply by nspsck** [reply]
Yes, it is necessary.

**Reply by alsor62** [reply]
Thanks, I understand.
